src/src/com/microsoft/aad/adal/AuthenticationParameters.java

```diff
@@ -166,17 +167,17 @@ public static void createFromResourceUrl(Context context, final URL resourceUrl,
             public void run() {
                 HashMap<String, String> headers = new HashMap<String, String>();
                 headers.put(WebRequestHandler.HEADER_ACCEPT, WebRequestHandler.HEADER_ACCEPT_JSON);
-                HttpWebResponse webResponse = sWebRequest.sendGet(resourceUrl, headers);
 
-                if (webResponse != null) {
-                    if(webResponse.getResponseException() != null) {
-                        onCompleted(webResponse.getResponseException(), null);
-                    }
+                HttpWebResponse webResponse = null;
+                try {
+                    webResponse = sWebRequest.sendGet(resourceUrl, headers);
                     try {
                         onCompleted(null, parseResponse(webResponse));
                     } catch (ResourceAuthenticationChallengeException exc) {
                         onCompleted(exc, null);
```

Comment:
If we fail on parseResponse with ResourceAuthenticationChallengeException we won't invoke success callback,
but if we invoke success it means we didn't throw an exception and we won't invoke callback for error